zetaclient/zetabridge.go

```diff
@@ -170,33 +172,84 @@ func (b *ZetaCoreBridge) UpdateConfigFromCore(cfg *config.Config) error {
 	if err != nil {
 		return err
 	}
-	chains := make([]common.Chain, len(coreParams))
+	newChains := make([]common.Chain, len(coreParams))
+	newEVMParams := make(map[int64]*types.CoreParams)
+	var newBTCParams *types.CoreParams
+
+	// check and update core params for each chain
 	for i, params := range coreParams {
-		chains[i] = *common.GetChainFromChainID(params.ChainId)
-		if common.IsBitcoinChain(params.ChainId) {
-			if cfg.BitcoinConfig == nil {
-				panic("BitcoinConfig is nil for this client")
-			}
-			if cfg.BitcoinConfig.CoreParams == nil {
-				cfg.BitcoinConfig.CoreParams = config.NewCoreParams()
-			}
-			cfg.BitcoinConfig.CoreParams.UpdateCoreParams(params)
-			continue
+		err := checkCoreParams(params)
+		if err != nil {
+			return err
 		}
-		_, found := cfg.EVMChainConfigs[params.ChainId]
-		if !found {
-			panic(fmt.Sprintf("EvmConfig %d is nil for this client ", params.ChainId))
-		}
-		if cfg.EVMChainConfigs[params.ChainId].CoreParams == nil {
-			cfg.EVMChainConfigs[params.ChainId].CoreParams = config.NewCoreParams()
+		newChains[i] = *common.GetChainFromChainID(params.ChainId)
+		if common.IsBitcoinChain(params.ChainId) {
+			newBTCParams = params
+		} else {
+			newEVMParams[params.ChainId] = params
 		}
-		cfg.EVMChainConfigs[params.ChainId].CoreParams.UpdateCoreParams(params)
 	}
-	cfg.ChainsEnabled = chains
 	keyGen, err := b.GetKeyGen()
 	if err != nil {
 		return err
 	}
-	cfg.Keygen = *keyGen
+	cfg.UpdateCoreParams(keyGen, newChains, newEVMParams, newBTCParams, init)
+
+	return nil
+}
+
+// Some basic checks on core params
+func checkCoreParams(coreParams *types.CoreParams) error {
```

Comment:
The convention in Cosmos SDK for type validation is generally `Validate<typeName>` I think we could name the method `ValidateCoreParams`